src/explore-education-statistics-common/src/components/CopyLinkButton.tsx

```diff
@@ -1,51 +1,80 @@
 import Button from '@common/components/Button';
-import LinkIcon from '@common/components/LinkIcon';
 import styles from '@common/components/CopyLinkButton.module.scss';
-import Modal from '@common/components/Modal';
+import { CopyLinkModalProps } from '@common/components/CopyLinkModal';
 import UrlContainer from '@common/components/UrlContainer';
-import VisuallyHidden from '@common/components/VisuallyHidden';
 import useToggle from '@common/hooks/useToggle';
 import classNames from 'classnames';
-import React from 'react';
+import { OmitStrict } from '@common/types';
+import React, { ReactNode } from 'react';
 
-interface Props {
-  className?: string;
-  url: string;
+interface Props extends OmitStrict<CopyLinkModalProps, 'buttonClassName'> {
+  buttonText?: string | ReactNode;
+  label?: string;
 }
 
-const CopyLinkButton = ({ className, url }: Props) => {
+export default function CopyLinkButton({
+  buttonText = 'Copy',
+  className,
+  confirmMessage = 'Link copied to the clipboard.',
+  inlineButton = true,
+  label,
+  url,
+}: Props) {
   const [copied, toggleCopied] = useToggle(false);
 
   return (
-    <Modal
-      showClose
-      title="Copy link to the clipboard"
-      triggerButton={
-        <Button className={className} variant="secondary">
-          <LinkIcon />
-          <VisuallyHidden>Copy link to the clipboard</VisuallyHidden>
-        </Button>
-      }
-    >
+    <div className={className}>
```

Comment:
Do we need the extra div? Can't this be combined with the child container div?

Reply:
It provides a container for the whole component so you can do things like add margin around it, the child container div doesn't include the message when it's inline.